### DISCUSSION TITLE
Browser repaint issue

### DISCUSSION THREAD

**Discussion by ekalchev** [cefsharp/CefSharp#4088] [General]
- **What architecture x86 or x64?**
    x64
	
- **What version of .Net?**
    .Net 4.8
    
- **On what operating system?**
    Win11

- **Are you using `WinForms`, `WPF` or `OffScreen`?**
    WinForms with WindowsFormsHost in WPF app

I just upgraded from CefSharp 81 to 100 and I noticed delay of browser repaint after the page is loaded. 
![Alt Text](http://g.recordit.co/VW2GGQFs8V.gif)

Here gif that shows the issue. This is mail app and I am switching between emails. On the right in the debugger you can clearly see that DOM is loaded fast but on the right in the browser the image of the old page is visible for a while. 

I was able to capture the issue in the performance tab. You can clearly see DOM Loaded occurred at ~190ms but the repaint happened at ~990ms

![image](https://user-images.githubusercontent.com/10009149/166514683-c20dd4d5-a1fa-4e45-b283-e9a1ffac6026.png)

I rolled back to version 81 and this is how fast is repainted there

![image](https://user-images.githubusercontent.com/10009149/166516944-f8af7b19-fee0-4057-8de1-7557cf86df3f.png)

Any suggestions what could cause this? Is there any setting that I can turn on/off to bring the old behavior?

**Comment by amaitland** [comment]
Converted to discussion as this is unlikely a bug in CefSharp. 

What troubleshooting have you done already?

Check the log file? 

What command line args are you using if any?

**Comment by ekalchev** [comment]
I was able to reproduce the issue in minimal example. It seems the issue is not present in version 96.0.180 and appear in next version 97.1.11

Steps to reproduce:

1. Start Winforms example
2. Wait for start page to appear
3. Click Go button - this will load html page (Delivery has failed page)
4. Click Again Go Button - this time empty page should be loaded

Expected: Empty page should be loaded immediately.
Actual: There is slight delay in version 97.1.11 compared to version 96.0.180 before the empty page is displayed

I prepared two commits in this fork where you can see it
https://github.com/ekalchev/CefSharp.MinimalExample

First commit is with version 96 where it works fine, second commit bump the version to 97 and you can see the delay with naked eye

Initially I tried to reproduce this without ResourceHandler with LoadHtml method but I couldn't. Once I created the TestResourceHandler this became reproducible.

**Reply by amaitland** [reply]
Have you tested on different hardware? 

Did you test with the CEF Sample application? 

Did you check the log file?

**Reply by ekalchev** [reply]
Yeah, I am reproducing on different machines (3 machines with different hardware Win11 and Win10)

No, I haven't test it on CEF Sample application

Yes, I can't see anything wrong in the log file

**Reply by amaitland** [reply]
> Actual: There is slight delay in version 97.1.11 compared to version 96.0.180 before the empty page is displayed

You can bisect the 97 branch

https://github.com/cefsharp/CefSharp/commits/cefsharp/97

**Comment by ekalchev** [comment]
I found exact commit where this issue become reproducible 

this is last commit where the issue is not reproducible 
8160ec1ce4bef9b9289157ac1fd26aa2bc3f06c2

And after two commits the issue is reproducible
4313c9a25f657f8d6cb55b949dd5765477dc6240

I can see between these two commit cef-runtime was updated to v97. I assume this is upstream issue?

**Comment by ekalchev** [comment]
I was able to reproduce it with CEF Sample application
https://bitbucket.org/chromiumembedded/cef/issues/3326/slower-browser-repaint-after-cef-v96

**Comment by ekalchev** [comment]
This was accepted as bug from Chromium team. For workaround for this issue you can use `--disable-features=PaintHoldingCrossOrigin` or `--disable-features=PaintHolding`
